### ISSUE TITLE
`NoReturn` becomes `Any` when function is asynchronous

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1182]
**Describe the bug**
An asynchronous function which returns `NoReturn` has its return type replaced with `Any`.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That the type of `bar` would be `() -> Coroutine[Any, Any, NoReturn]`.

**Screenshots or Code**
```python
# pyright: strict

from __future__ import annotations

from typing import NoReturn

def foo() -> NoReturn:
    ...

async def bar() -> NoReturn:
    ...

def baz() -> None:
    reveal_type(foo)  # Type of "foo" is "() -> NoReturn"
    reveal_type(foo())  # Type of "foo()" is "NoReturn"
    'I am baz'  # Unreachable

async def qux() -> None:
    reveal_type(bar)  # Type of "bar" is "() -> Coroutine[Any, Any, Any]"
    reveal_type(bar())  # Type of "bar()" is "Coroutine[Any, Any, Any]"
    reveal_type(await bar())  # Type of "await bar()" is "Any"
    'I am qux'  # Reachable

```

**VS Code extension or command-line**
Either, 1.1.86

**Comment by erictraut** [comment]
The "Any" behavior was added in response to this bug report: https://github.com/microsoft/pylance-release/issues/325, but upon a second look, I don't think that was the correct solution. I've removed the code that translates NoReturn to Any for the async case.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.87, which I just published. It will also be included in the next release of pylance.
